bookwyrm/models/bookwyrm_export_job.py

```diff
@@ -63,138 +63,150 @@ def tar_export(json_data: str, user, file):
+    # User as AP object
+    exported_user = user.to_activity()
+    # I don't love this but it prevents a JSON encoding error
+    # when there is no user image
+    if isinstance(exported_user["icon"], dataclasses._MISSING_TYPE): # pylint: disable=protected-access
+        exported_user["icon"] = {}
+    else:
+        # change the URL to be relative to the JSON file
+        file_type = exported_user["icon"]["url"].rsplit(".", maxsplit=1)[-1]
+        filename = f"avatar.{file_type}"
+        exported_user["icon"]["url"] = filename
+
+    # Additional settings
+    # can't be serialized as AP
     vals = [
-        "username",
-        "name",
-        "summary",
-        "manually_approves_followers",
-        "hide_follows",
         "show_goal",
-        "show_suggested_users",
-        "discoverable",
         "preferred_timezone",
         "default_post_privacy",
+        "show_suggested_users",
     ]
+    exported_user["settings"] = {}
     for k in vals:
-        exported_user[k] = getattr(user, k)
+        exported_user["settings"][k] = getattr(user, k)
 
-    if getattr(user, "avatar", False):
-        exported_user["avatar"] = f'https://{DOMAIN}{getattr(user, "avatar").url}'
-
-    # reading goals
+    # Reading goals
+    # can't be serialized as AP
     reading_goals = AnnualGoal.objects.filter(user=user).distinct()
-    goals_list = []
-    # TODO: either error checking should be more sophisticated
-    # or maybe we don't need this try/except
-    try:
-        for goal in reading_goals:
-            goals_list.append(
-                {"goal": goal.goal, "year": goal.year, "privacy": goal.privacy}
-            )
-    except Exception:  # pylint: disable=broad-except
-        pass
+    exported_user["goals"] = []
+    for goal in reading_goals:
+        exported_user["goals"].append(
+            {"goal": goal.goal, "year": goal.year, "privacy": goal.privacy}
+        )
+
+    # Reading history
+    # can't be serialized as AP
+    readthroughs = ReadThrough.objects.filter(user=user).distinct().values()
+    readthroughs = list(readthroughs)
+
+    # Books
+    editions = get_books_for_user(user)
+    exported_user["books"] = []
+    for edition in editions:
+        book = {}
+        book["edition"] = edition.to_activity()
 
-    try:
-        readthroughs = ReadThrough.objects.filter(user=user).distinct().values()
-        readthroughs = list(readthroughs)
-    except Exception:  # pylint: disable=broad-except
-        readthroughs = []
-
-    # books
-    editions, books = get_books_for_user(user)
-    final_books = []
-
-    for book in books.values():
-        edition = editions.filter(id=book["id"])
-        book["edition"] = edition.values()[0]
         # authors
-        book["authors"] = list(edition.first().authors.all().values())
-        # readthroughs
+        book["authors"] = []
+        for author in edition.authors.all():
+            obj = author.to_activity() # <== this doesn't include blank optional fields
```

Comment:
`to_activity` here drops any fields that are `Optional` and null. This makes sense but `to_model` on import expects them to exist, so throws and exception